src/NHibernate.Test/PropertyRef/ManyToManyPropertyRefFixture.cs

```diff
@@ -0,0 +1,137 @@
+using NHibernate.Criterion;
+using NUnit.Framework;
+
+namespace NHibernate.Test.PropertyRef
+{
+	[TestFixture]
+	public class ManyToManyPropertyRefFixture : TestCase
+	{
+		protected override string[] Mappings => new[] { "PropertyRef.ManyToManyWithPropertyRef.hbm.xml" };
+
+		protected override string MappingsAssembly => "NHibernate.Test";
+
+		private object _manyAId;
+
+		protected override void OnSetUp()
+		{
+			using (var session = OpenSession())
+			using (var transaction = session.BeginTransaction())
+			{
+				var manyA = new ManyA { Number = 6, Value = "a value" };
+
+				var manyB1 = new ManyB { Number = 4, Value = "a value of b1" };
+				var manyB2 = new ManyB { Number = 8, Value = "a value of b2" };
+				var manyB3 = new ManyB { Number = 12, Value = "a value of b3" };
+
+				_manyAId = session.Save(manyA);
+				session.Save(manyB1);
+				session.Save(manyB2);
+				session.Save(manyB3);
+
+				manyA.ManyBs.Add(manyB1);
+				manyA.ManyBs.Add(manyB2);
+				manyA.ManyBs.Add(manyB3);
+				transaction.Commit();
+			}
+		}
+
+		protected override void OnTearDown()
+		{
+			using (var s = OpenSession())
+			using (var t = s.BeginTransaction())
+			{
+				s.Delete("from System.Object");
+				s.Flush();
+				t.Commit();
+			}
+		}
+
+		[Test]
+		public void Getting_a_ManyA_object_with_fetchmode_select_will_work()
+		{
+			ManyA loadedManyA;
+			using (var session = OpenSession())
+			using (var transaction = session.BeginTransaction())
+			{
+				loadedManyA =
+					session
+						.CreateCriteria<ManyA>()
+						.Add(Restrictions.IdEq(_manyAId))
+						// Below Fetch is a no-op indeed, provided the mapping does not ask for eager fetching.
+						// It is the equivalent of obsoleted SetFetchMode Select, which was also no-op, contrary
+						// to what the SelectMode xml comment could let think. (This comment was valid only when
+						// the enum was used for mapping, but not when used ins queries.)
+						.Fetch(SelectMode.Skip, "ManyBs")
+						.UniqueResult<ManyA>();
+				NHibernateUtil.Initialize(loadedManyA.ManyBs);
+				transaction.Commit();
+			}
+
+			/******************the select statements *************************************************************
+			SELECT	this_.Id as Id0_0_,
+					this_.Number as Number0_0_,
+					this_.Value as Value0_0_ 
+			FROM	ManyA this_ 
+			WHERE	this_.Id = 1 /# ?p0 /#
+
+			SELECT	manybs0_.ManyBNumber as ManyBNum1_1_,
+					manybs0_.ManyANumber as ManyANum2_1_,
+					manyb1_.Id           as Id2_0_,
+					manyb1_.Number       as Number2_0_,
+					manyb1_.Value        as Value2_0_
+			FROM	ManyBs manybs0_
+			left outer join ManyB manyb1_ on manybs0_.ManyANumber = manyb1_.Number
+			WHERE  manybs0_.ManyBNumber =6 /# ?p0 #/
+			 */
+
+			Assert.That(loadedManyA.ManyBs.Count, Is.EqualTo(3));
+		}
+
+		[Test, Ignore("Not fixed yet")]
+		public void Getting_a_ManyA_object_with_fetchmode_join_will_work()
```

Comment:
This test case is originally meant for #1214. This test here is now failing again like initially reported. #1214 cannot be fixed without first fixing the regression caused by #378.
The other test was not failing in the original context of #1214, and is the one demonstrating #378 regression.